src/core/toga/widgets/canvas.py

```diff
@@ -393,6 +393,7 @@ def _draw(self, impl, *args, **kwargs):
         """
         impl.new_path(*args, **kwargs)
         for obj in self.drawing_objects:
+            kwargs["fill_color"] = self.color
```

Comment:
I take it this is required because GTK drawing objects are "contextual", and so retain their ancestors fill color, but Cocoa has independent drawing APIs that need all the detail provided at every level?

Reply:
Thanks for your review, I really appreciate your time :+1: 

That's right. Gtk uses the normal fill and stroke methods to fill and stroke all drawing methods, including text. Cocoa has separate methods to change text color and draw the text. This means that if we create a toga Fill context, and in that context we create a rectangle and draw text, the color applied to the Fill context and the Fill operation won't fill the text as well without us manually passing in the color to the write_text implementation.

Since we are calling obj._draw(impl) for all drawing objects, I had to either pass *args and **kwargs to every implementation draw operation, or check that the obj is an instance of the WriteText class and only then pass in the color. I tried it both ways, and I thought that using the first method was much more clean.